src/Concordium/Client/Types/Contract/Schema.hs

```diff
@@ -486,7 +553,15 @@ instance S.Serialize SchemaType where
       28 -> S.label "ILeb128"  $ ILeb128 <$> S.getWord32le
       29 -> S.label "ByteList" $ ByteList <$> S.get
       30 -> S.label "ByteArray"  $ ByteArray <$> S.getWord32le
+      31 -> S.label "TaggedEnum" $ TaggedEnum
+        <$> getMapOfWithSizeLenAndPred tEnumPred Four S.getWord8 (S.getTwoOf getText S.get)
       x  -> fail [i|Invalid SchemaType tag: #{x}|]
+    where
+      -- Predicate for tagged enums. Tags and variant names should be unique.
+      allUnique :: (Ord a) => [a] -> Bool
+      allUnique xs = length (nub xs) == length xs
```

Comment:
Nub is quadratic, but you required a to be comparable. So use sort sort + group instead.

Reply:
True, but I thought this was much more readable since the asymptotics don't matter much here (as n <= 256). I'll change it.